frappe/utils/csvutils.py

```diff
@@ -39,7 +40,7 @@ def read_csv_content_from_attached_file(doc):
 def read_csv_content(fcontent):
 	if not isinstance(fcontent, str):
 		decoded = False
-		for encoding in ["utf-8", "windows-1250", "windows-1252"]:
+		for encoding in ["utf-8-sig", "utf-8", "windows-1250", "windows-1252"]:
```

Comment:
Can store these as a separate list and import into file/file.py and here so that it needs to be maintained in only one place (maybe frappe/utils/file_manager.py - not sure, maybe there's somewhere else more appropriate too)

Reply:
@akhilnarang I dig through the code a little bit, I didn't find any good place to store these kind of "constants", only maybe in the root frappe init file (frappe/__init__.py), around line 58, but this might feel a bit overkill.
``` python
controllers = {}
local = Local()
cache = None
STANDARD_USERS = ("Guest", "Administrator")
FILE_ENCODING_OPTIONS = ("utf-8-sig", "utf-8", "windows-1250", "windows-1252")
_qb_patched = {}
_dev_server = int(sbool(os.environ.get("DEV_SERVER", False)))
_tune_gc = bool(sbool(os.environ.get("FRAPPE_TUNE_GC", True)))
```

Reply:
nvm, I moved it to file.py, can be imported by:
`from frappe.core.doctype.file.file import FILE_ENCODING_OPTIONS`